File: ch5/akkademaid-scala/src/main/scala/com/akkademy/ArticleParseActor.scala
package com.akkademy

import akka.actor.Actor

class ArticleParseActor extends Actor{
  override def receive: Receive = {
    case ParseArticle(htmlString) =>
      val body: String = ArticleParser(htmlString)
      sender() ! body
  }
}

object ArticleParser {
  def apply(html: String) : String =
    de.l3s.boilerpipe.extractors.ArticleExtractor.INSTANCE.getText(html)
}

File: ch5/akkademaid-java/src/main/java/akkademy/ArticleParseActor.java
package akkademy;

import akka.actor.AbstractActor;
import akka.actor.Status;
import akka.japi.pf.ReceiveBuilder;

public class ArticleParseActor extends AbstractActor {
  private ArticleParseActor() {
    receive(ReceiveBuilder.
        match(ParseArticle.class, x ->{
                    ArticleParser.apply(x.htmlBody).
                            onSuccess(body -> sender().tell(body, self())).
                            onFailure(t -> sender().tell(new Status.Failure(t), self()));
                }
        ).
        build());
  }
}
